programs/local/LocalServer.cpp

```diff
@@ -967,6 +967,14 @@ void LocalServer::processConfig()
         attachSystemTablesServer(global_context, *createMemoryDatabaseIfNotExists(global_context, DatabaseCatalog::SYSTEM_DATABASE), false);
         attachInformationSchema(global_context, *createMemoryDatabaseIfNotExists(global_context, DatabaseCatalog::INFORMATION_SCHEMA));
         attachInformationSchema(global_context, *createMemoryDatabaseIfNotExists(global_context, DatabaseCatalog::INFORMATION_SCHEMA_UPPERCASE));
+
+        // Background tasks are required for DDL operations like DROP VIEW SYNC, even in temporary mode (no --path).
+        // Only skip if --only-system-tables is requested.
+        if (!getClientConfiguration().has("only-system-tables"))
```

Comment:
We are currently also inside a `!getClientConfiguration().has("no-system-tables")` condition (see line 965), looks like we should also add it to `else` after it.

Reply:
Yeah, makes sense. Made the fix, now it should work correctly with both `--only-system-tables` and `--no-system-tables`.